pkg/okteto/pipeline.go

```diff
@@ -43,28 +42,73 @@ func newPipelineClient(client *graphql.Client, url string) *pipelineClient {
 	}
 }
 
+type deployPipelineMutation struct {
+	Response deployPipelineResponse `graphql:"deployGitRepository(name: $name, repository: $repository, space: $space, branch: $branch, variables: $variables, filename: $filename)"`
+}
+
+type getPipelineByNameQuery struct {
+	Response getPipelineByNameResponse `graphql:"space(id: $id)"`
+}
+
+type destroyPipelineWithVolumesMutation struct {
+	Response destroyPipelineResponse `graphql:"destroyGitRepository(name: $name, space: $space, destroyVolumes: $destroyVolumes)"`
+}
+
+type deprecatedDestroyPipelineWithVolumesMutation struct {
+	Response deprecatedDestroyPipelineResponse `graphql:"destroyGitRepository(name: $name, space: $space, destroyVolumes: $destroyVolumes)"`
+}
+
+type destroyPipelineWithoutVolumesMutation struct {
+	Response destroyPipelineResponse `graphql:"destroyGitRepository(name: $name, space: $space)"`
+}
+
+type deprecatedDestroyPipelineWithoutVolumesMutation struct {
+	Response deprecatedDestroyPipelineResponse `graphql:"destroyGitRepository(name: $name, space: $space)"`
+}
+
+type getPipelineResources struct {
+	Response previewResourcesStatus `graphql:"space(id: $id)"`
+}
+
+type deployPipelineResponse struct {
+	Action    actionStruct
+	GitDeploy gitDeployInfoWithRepoInfo
+}
+
+type gitDeployInfoWithRepoInfo struct {
+	Id         graphql.String
+	Name       graphql.String
+	Status     graphql.String
+	Repository graphql.String
+}
+
+type getPipelineByNameResponse struct {
+	GitDeploys []gitDeployInfoIdNameStatus
+}
+
+type gitDeployInfoIdNameStatus struct {
+	Id     graphql.String
+	Name   graphql.String
+	Status graphql.String
+}
+
+type destroyPipelineResponse struct {
+	Action    actionStruct
+	GitDeploy gitDeployInfoWithRepoInfo
+}
+
+type deprecatedDestroyPipelineResponse struct {
+	GitDeploy gitDeployInfoWithRepoInfo
+}
+
 // Deploy creates a pipeline
 func (c *pipelineClient) Deploy(ctx context.Context, opts types.PipelineDeployOptions) (*types.GitDeployResponse, error) {
 	origin := config.GetDeployOrigin()
 
 	gitDeployResponse := &types.GitDeployResponse{}
 
 	if len(opts.Variables) > 0 {
-		var mutation struct {
-			GitDeployResponse struct {
-				Action struct {
-					Id     graphql.String
-					Name   graphql.String
-					Status graphql.String
-				}
-				GitDeploy struct {
-					Id         graphql.String
-					Name       graphql.String
-					Status     graphql.String
-					Repository graphql.String
-				}
-			} `graphql:"deployGitRepository(name: $name, repository: $repository, space: $space, branch: $branch, variables: $variables, filename: $filename)"`
-		}
+		var mutation deployPipelineMutation
```

Comment:
```suggestion
		var deployMutation deployPipelineMutation
```
I think it improves the reading to add more info about the mutation.